### ISSUE TITLE
[A11y] Revision Card

### ISSUE THREAD

**Issue by rt4914** [oppia/oppia-android#2637]
## Current Output
https://user-images.githubusercontent.com/9396084/107035557-587e3980-67de-11eb-92de-57b503f78d18.mp4

## Issues Identified
- [x] The talkback reads this screen as **Oppia** at start, instead it should be **Skill page**. Also please add test for the same. (Assigned to @jonathanalvares9009)
- [ ] After back arrow, overflow menu is selected and "What is a Fraction?" is never selected. Order should be: back-arrow, "Title" and overflow menu. (Assigned to @rt4914)

Note: To understand the above issue(s) completely, it is recommended that you setup Talkback and play with the app keeping it on and that will give you better context.

**Accessibility Guide**: https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide

**Comment by Prince-kushwaha** [comment]
i want to work on this issue

**Comment by rt4914** [comment]
@Prince-kushwaha I have assigned you two issues right now. So not assigning you this issue for now. Let's wait for your other PRs to get merged.

**Comment by rioX432** [comment]
@rt4914  I want to work on this issue.
If it is ok, please assign this task to me.

**Comment by rt4914** [comment]
@omooooori Thanks for showing interest. Please checkout our [wiki](https://github.com/oppia/oppia-android/wiki) first. This will help you in setting up the project and once you sign the CLA and fill the survey form we can assign you issues to work on.

**Comment by priyanka0906** [comment]
@rt4914  may I work on this?

**Comment by rioX432** [comment]
@rt4914  I signed CLA and answered survey. So if it is ok, could you please assign this task to me.

**Comment by Akshat-gour** [comment]
@rt4914 I tested this app on physical device (Samsung C7 pro API 26)  as well as on emulator(Pixel 2 API 29) 
Talkback works in this order Oppia ,back-arrow, "Title" and overflow menu.



But Only Read Oppia once when we open the what is fraction under Revision Card .  

Here is One More Issue when this Sound icon is tapped talkback says Unlabelled
![Capture 1](https://user-images.githubusercontent.com/60266468/107154445-ff0e3a00-6998-11eb-9705-5edc3f71e7b5.PNG)

**Comment by rt4914** [comment]
> @rt4914 I want to work on this issue.
> If it is ok, please assign this task to me.

@omooooori Assigned you one part of the issue.

**Comment by rt4914** [comment]
> @rt4914 may I work on this?

@priyanka0906 Assigned you one part of the issue.

**Comment by rt4914** [comment]
> @rt4914 I tested this app on physical device (Samsung C7 pro API 26) as well as on emulator(Pixel 2 API 29)
> Talkback works in this order Oppia ,back-arrow, "Title" and overflow menu.
> 
> But Only Read Oppia once when we open the what is fraction under Revision Card .
> 
> Here is One More Issue when this Sound icon is tapped talkback says Unlabelled
> ![Capture 1](https://user-images.githubusercontent.com/60266468/107154445-ff0e3a00-6998-11eb-9705-5edc3f71e7b5.PNG)

@Akshat-gour Thanks for checking. What you are trying out is an exploration and this issue is related to `Revision Card`. You can reach revision card by following these steps: Home Screen → Select Fractions → Select `Revision` tab → Select first card.

**Comment by Akshat-gour** [comment]
I tried Revision also
![1](https://user-images.githubusercontent.com/60266468/107184517-b8efbf80-6a06-11eb-8bad-49a2c91e3fe1.PNG)

**Comment by rt4914** [comment]
> I tried Revision also
> ![1](https://user-images.githubusercontent.com/60266468/107184517-b8efbf80-6a06-11eb-8bad-49a2c91e3fe1.PNG)

@Akshat-gour It should not read Oppia at all instead it should read only the title of the screen.

**Comment by rioX432** [comment]
@rt4914 Sorry. I have simple question.
How can we put the talkback on button ?
I would like to switch talkback on/off by on click.

![image](https://user-images.githubusercontent.com/34413373/107849097-3a4ab600-6e3c-11eb-8e75-d6f8adbc28d3.png)

**Comment by Arjupta** [comment]
> @rt4914 Sorry. I have simple question.
> How can we put the talkback on button ?
> I would like to switch talkback on/off by on click.
> 
> ![image](https://user-images.githubusercontent.com/34413373/107849097-3a4ab600-6e3c-11eb-8e75-d6f8adbc28d3.png)

@omooooori the instructions varies for different devices. You can search about enabling TalkBack in a device which you use. There are some shortcuts combination of volume and power button also( again specific for different devices)

**Comment by rt4914** [comment]
@omooooori I have just updated the assigned issue which makes it much simple. You just need a static label instead of a dynamic label.

**Comment by priyanka0906** [comment]
@rt4914  in my emulator ,talkback is not navigating to all the items, as shown in the video.

https://user-images.githubusercontent.com/56197993/108250998-ef52da80-717c-11eb-9939-0ee408d10553.mp4

**Comment by Akshat-gour** [comment]
@priyanka0906 
Talkback Does not Automatically Read the next text.
Take a look at this to know how navigation works.
https://support.google.com/accessibility/android/answer/6006598
also
https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide

**Comment by priyanka0906** [comment]
> @priyanka0906
> Talkback Does not Automatically Read the next text.
> Take a look at this to know how navigation works.
> https://support.google.com/accessibility/android/answer/6006598

okay...thankyou

**Comment by priyanka0906** [comment]
@rt4914  I have tried   android:accessibilityTraversalAfter="@id/revision_card_toolbar"  but it didn't work....would you please provide any hint or something which would help me.

**Comment by rt4914** [comment]
> @rt4914 I have tried android:accessibilityTraversalAfter="@id/revision_card_toolbar" but it didn't work....would you please provide any hint or something which would help me.

@priyanka0906 Suggest checking out `Topic` and `StoryActivity` toolbar and how it is implemented if we follow similar approach. It should work in `Revision Card `too.

**Comment by priyanka0906** [comment]
> > @rt4914 I have tried android:accessibilityTraversalAfter="@id/revision_card_toolbar" but it didn't work....would you please provide any hint or something which would help me.
> 
> @priyanka0906 Suggest checking out `Topic` and `StoryActivity` toolbar and how it is implemented if we follow similar approach. It should work in `Revision Card `too.

okay thankyou

**Comment by priyanka0906** [comment]
@rt4914 in this also title is not selected after back arrow

https://user-images.githubusercontent.com/56197993/108584880-bed09380-736a-11eb-80b6-fe735ac78df5.mp4

**Comment by rt4914** [comment]
@priyanka0906 Thanks for providing the video. I think I will have to try to solve this and then only I can confirm the exact reason for the issue.

**Comment by priyanka0906** [comment]
> @priyanka0906 Thanks for providing the video. I think I will have to try to solve this and then only I can confirm the exact reason for the issue.

okay sir

**Comment by Sparsh1212** [comment]
@rt4914 If the first part of the issue is still available, then please assign it to me. Also,  please tell the static `label` for this screen.

**Comment by rt4914** [comment]
@Sparsh1212 First part needs confirmation on another issue from Chantel. Also, I suggest not picking these issues as you have already finished multiple issues where you are adding label. Instead try to pick something new.

**Comment by Sparsh1212** [comment]
> @Sparsh1212 First part needs confirmation on another issue from Chantel. Also, I suggest not picking these issues as you have already finished multiple issues where you are adding label. Instead try to pick something new.

Sure, I have commented on other different issues as well, will start working on them as assigned.

**Comment by rioX432** [comment]
> @omooooori I have just updated the assigned issue which makes it much simple. You just need a static label instead of a dynamic label.

@rt4914 I tried to resolve according to your commnet.

I think `RevisionCardActivityPresenter::subscribeToSubtopicTitle()` is creating sub topic title dynamically.
But if I change not to use this and set static sub topic title, all revision card sub topic title will be changed.
You mean that we should change to set static label we got from json file to `revisionCardToolbarTitle.text` ?

Or you mean we have to not to use LiveData to set sub topic title ?
I would like to get some advise.

Thank you in advance.

**Comment by rt4914** [comment]
> > @omooooori I have just updated the assigned issue which makes it much simple. You just need a static label instead of a dynamic label.
> 
> @rt4914 I tried to resolve according to your commnet.
> 
> I think `RevisionCardActivityPresenter::subscribeToSubtopicTitle()` is creating sub topic title dynamically.
> But if I change not to use this and set static sub topic title, all revision card sub topic title will be changed.
> You mean that we should change to set static label we got from json file to `revisionCardToolbarTitle.text` ?
> 
> Or you mean we have to not to use LiveData to set sub topic title ?
> I would like to get some advise.
> 
> Thank you in advance.

@omooooori I have made this issue simpler. You can introduce a static string which reads `Skill page` instead of dynamic string.

**Comment by jonathanalvares9009** [comment]
@rt4914 I would like to take this issue if it's not yet complete can I get a little more context on what I'll have to do?

**Comment by jonathanalvares9009** [comment]
@rt4914 I'm trying to solve part 1 bug. I used the app with talkback. So when I click on something to raise it says oppia n then the name of the unit eg: What is a fraction?. I have to replace oppia with skill page right don't touch the header of the revision card n also write test for the implementation. Please correct if I've misunderstood the issue.

**Comment by jonathanalvares9009** [comment]
@rt4914 sir I wrote the code that says skill page instead of oppia. Do I have to write a test for the same or is the change is enough?
